// rete-angular-render-plugin/esm2020/module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { createCustomElement } from '@angular/elements';
import { KebabPipe } from './kebab.pipe';
import { NodeComponent } from './node/node.component';
import { SocketComponent } from './socket/socket.component';
import { ControlDirective } from './control.directive';
import { SocketDirective } from './socket.directive';
import { CustomComponent } from './custom.component';
import * as i0 from "@angular/core";
export class ReteModule {
    constructor(injector) {
        const CustomElement = createCustomElement(CustomComponent, { injector });
        if (!customElements.get('rete-element'))
            customElements.define('rete-element', CustomElement);
    }
}
ReteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.2", ngImport: i0, type: ReteModule, deps: [{ token: i0.Injector }], target: i0.ɵɵFactoryTarget.NgModule });
ReteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.1.2", ngImport: i0, type: ReteModule, declarations: [NodeComponent,
        SocketComponent,
        CustomComponent,
        ControlDirective,
        SocketDirective,
        KebabPipe], imports: [CommonModule], exports: [NodeComponent,
        CustomComponent,
        SocketComponent,
        ControlDirective,
        SocketDirective,
        KebabPipe] });
ReteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.1.2", ngImport: i0, type: ReteModule, providers: [
        KebabPipe,
        ControlDirective
    ], imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.2", ngImport: i0, type: ReteModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NodeComponent,
                        SocketComponent,
                        CustomComponent,
                        ControlDirective,
                        SocketDirective,
                        KebabPipe,
                    ],
                    imports: [
                        CommonModule
                    ],
                    providers: [
                        KebabPipe,
                        ControlDirective
                    ],
                    exports: [
                        NodeComponent,
                        CustomComponent,
                        SocketComponent,
                        ControlDirective,
                        SocketDirective,
                        KebabPipe
                    ],
                    entryComponents: [
                        NodeComponent,
                        SocketComponent,
                        CustomComponent
                    ]
                }]
        }], ctorParameters: function () { return [{ type: i0.Injector }]; } });
